pkg/schema/type_check.go

```diff
@@ -0,0 +1,113 @@
+package schema
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/authzed/spicedb/pkg/genutil/mapz"
+	corev1 "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+const ellipsesRelation = "..."
+
+func (ts *TypeSystem) GetTypesForRelation(ctx context.Context, defName string, relationName string) (*mapz.Set[string], error) {
```

Comment:
Let's name this `GetRecursiveSubjectTypesForRelation` and add a doc comment explaining what it returns. I'd also change the return to be a slice of names, rather than the set